lib/ret_web/plugs/proxies.ex

```diff
@@ -1,6 +1,20 @@
 defmodule RetWeb.Plugs.PostgrestProxy do
   use Plug.Builder
-  plug ReverseProxyPlug, upstream: "http://localhost:3000"
+
+  plug :call
+
+  @spec call(Plug.Conn.t(), []) :: Plug.Conn.t()
+  def call(conn, []) do
+    opts = ReverseProxyPlug.init(upstream: "http://#{hostname()}:3000")
+    ReverseProxyPlug.call(conn, opts)
+  end
+
+  @spec hostname :: String.t()
+  defp hostname,
+    do:
+      :ret
+      |> Application.fetch_env!(__MODULE__)
+      |> Keyword.fetch!(:hostname)
```

Comment:
Looking at this again, I wonder if we should use the same pattern we use elsewhere in the code. More of a nit, but it is a bit cleaner:
```suggestion
  defp hostname, do: Application.get_env(:ret, __MODULE__)[:hostname]
```

Reply:
`Application.get_env/2` and `Access` are permissive. If the application is misconfigured and either  `RetWeb.Plugs.PostgrestProxy` or `:hostname` don’t exist, a rogue `nil` is propagated through the call chain. `Application.fetch_env!/2` and `Keyword.fetch!/2` force the misconfigured application to fail within `hostname/0`.

Reply:
Ah, gotcha. Makes sense. Carry on.